MDEV-17091 - Assertion failed after dropping versioning

FooBarrior opened the pull request:
Assertion `old_part_id == m_last_part' failed in ha_partition::update_row or `part_id == m_last_part' in ha_partition::delete_row upon UPDATE/DELETE after dropping versioning

PRIMARY KEY change hadn't been treated as partition reorganization in case of partitioning by KEY() (without parameters).

* set `*partition_changed= true` in the described case.
* since add/drop system versioning does not affect alter_info->key_list, it required separate attention

an3l commented:
@FooBarrior  Is there any MDEV for this PR ?

claassistantio commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/MariaDB/server?pullRequest=1336) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/MariaDB/server?pullRequest=1336) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/MariaDB/server?pullRequest=1336) it.</sub>

an3l commented:
Hi @FooBarrior,
can you please rebase this PR and see what is happening with test `rpl.rpl_row_basic_8partition` which failed in [buildbot](http://buildbot.askmonty.org/buildbot/builders/kvm-rpm-centos6-amd64/builds/15135/steps/mtr/logs/stdio). 
Additionally, I think `delete` case is not covered in your test.

FooBarrior commented:
Thank you! I've rebased on fresh 10.3 and fixed the issue.

On Tue, 18 Feb 2020 at 11:37, Anel <notifications@github.com> wrote:

> Hi @FooBarrior <https://github.com/FooBarrior>,
> can you please rebase this PR and see what is happening with test
> rpl.rpl_row_basic_8partition which failed in buildbot
> <http://buildbot.askmonty.org/buildbot/builders/kvm-rpm-centos6-amd64/builds/15135/steps/mtr/logs/stdio>
> .
> Additionally, I think delete case is not covered in your test.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/MariaDB/server/pull/1336?email_source=notifications&email_token=AADAICKBEZAVOT5ZHZRD4C3RDM3UJA5CNFSM4HXWTLR2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEMAHPFI#issuecomment-587233173>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADAICO2TNGM2WJQCYHAXB3RDM3UJANCNFSM4HXWTLRQ>
> .
>


-- 
Yours truly,
Nikita Malyavin

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/MariaDB/server?pullRequest=1336) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/MariaDB/server?pullRequest=1336) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/MariaDB/server?pullRequest=1336) it.</sub>

FooBarrior commented:
@robertbindar how about this new one?

robertbindar commented:
@FooBarrior looks excellent but I meant the commit text, first line

FooBarrior commented:
@robertbindar sorry I can't. I've no write access to tempesta repo anymore. But I can press "squash and merge" and modify the commit message myself, or you can if you don't mind

robertbindar commented:
Many thanks @FooBarrior
